### PR TITLE
[SMALLFIX] Bug fixes for loadMetadata

### PR DESCRIPTION
**Pull request by madanadit**

### CODE REVIEW

File: core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java [hunk 1]
```diff
@@ -534,7 +534,8 @@ protected ObjectListingChunk getObjectListingChunkAndCreateNonEmpty(String path,
     String dir = stripPrefixIfPresent(path);
     ObjectListingChunk objs = getObjectListingChunk(dir, recursive);
     // If there are, this is a folder and we can create the necessary metadata
-    if (objs != null && objs.getObjectNames() != null && objs.getObjectNames().length > 0) {
+    if (objs != null && ((objs.getObjectNames() != null && objs.getObjectNames().length > 0)
+        || (objs.getCommonPrefixes() != null && objs.getCommonPrefixes().length > 0))) {
```
Comment (madanadit) [inline core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java]: There was a bug in which a directory with only subdirectories and no objects will not be discovered for s3a and likes. swift would have worked as it does not use commonPrefixes and combines prefixes in objectNames.

File: underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java [hunk 2]
```diff
@@ -319,6 +319,8 @@ protected ObjectListingChunk getObjectListingChunk(String key, boolean recursive
       throws IOException {
     String delimiter = recursive ? "" : PATH_SEPARATOR;
     key = PathUtils.normalizePath(key, PATH_SEPARATOR);
+    // In case key is root (empty string) do not normalize prefix
+    key = key.equals(PATH_SEPARATOR) ? "" : key;
```
Comment (madanadit) [inline underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java]: For listing root bucket, the prefix should be an empty string

**Comment by madanadit** [comment]
@ifcharming @calvinjia Please take a look!

**Comment by calvinjia** [comment]
LGTM, could you create a ticket for adding unit tests to validate these?

**Comment by ifcharming** [comment]
LGTM

**Pull request merged**